src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -1835,6 +2206,107 @@ static void Main()
+            CreateCompilation(source, parseOptions: TestOptions.Regular12).VerifyDiagnostics(expected);
+        }
+
+        [Fact]
+        public void BetterConversionFromExpression_10()
+        {
+            string source = """
+                using System;
+                using System.Collections.Generic;
+                class Program
+                {
+                    static void F2(List<int> value) { Console.WriteLine("List<int>"); }
+                    static void F2(List<byte> value) { Console.WriteLine("List<byte>"); }
+                    static void Main()
+                    {
+                        F2([(byte)1, (byte)2]);
+                    }
+                }
+                """;
+
+            CompileAndVerify(source, parseOptions: TestOptions.Regular13, expectedOutput: "List<byte>");
+
+            CreateCompilation(source, parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+                // (9,9): error CS0121: The call is ambiguous between the following methods or properties: 'Program.F2(List<int>)' and 'Program.F2(List<byte>)'
+                //         F2([1, (byte)2]);
+                Diagnostic(ErrorCode.ERR_AmbigCall, "F2").WithArguments("Program.F2(System.Collections.Generic.List<int>)", "Program.F2(System.Collections.Generic.List<byte>)").WithLocation(9, 9));
+        }
+
+        [Theory]
+        [InlineData("System.FormattableString")]
+        [InlineData("System.IFormattable")]
+        public void BetterConversionFromExpression_11(string formatType)
+        {
+            string source = $$"""
+                using System;
+                class Program
+                {
+                    static void F2(ReadOnlySpan<string> value) { Console.WriteLine("ReadOnlySpan<string>"); }
+                    static void F2(ReadOnlySpan<{{formatType}}> value) { Console.WriteLine("ReadOnlySpan<{{formatType}}>"); }
+                    static void Main()
+                    {
+                        F2([$"{1}"]);
+                    }
+                }
+                """;
+
+            CompileAndVerify(source,
+                parseOptions: TestOptions.Regular13,
+                targetFramework: TargetFramework.Net80,
+                verify: Verification.Skipped,
+                expectedOutput: IncludeExpectedOutput("ReadOnlySpan<string>"));
+
+            CreateCompilation(source, parseOptions: TestOptions.Regular12, targetFramework: TargetFramework.Net80).VerifyDiagnostics(
+                // (8,9): error CS0121: The call is ambiguous between the following methods or properties: 'Program.F2(ReadOnlySpan<string>)' and 'Program.F2(ReadOnlySpan<IFormattable>)'
+                //         F2([$"{1}"]);
+                Diagnostic(ErrorCode.ERR_AmbigCall, "F2").WithArguments("Program.F2(System.ReadOnlySpan<string>)", $"Program.F2(System.ReadOnlySpan<{formatType}>)").WithLocation(8, 9)
+                );
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Are we testing the following, where the collection types are distinct, one or both are span types, and there is an implicit numeric conversion from one element type to the other?
```csharp
using System;

class Program
{
    static void Main()
    {
        F1([1, 2, 3]); // C#12: F1(ReadOnlySpan<byte>); C#13: ambiguous
        F2([4, 5, 6]); // C#12: ambiguous; C#13: F2(ReadOnlySpan<int>)
    }
    
    static void F1(ReadOnlySpan<byte> arg) { }
    static void F1(Span<int> arg) { }
    
    static void F2(ReadOnlySpan<int> arg) { }
    static void F2(Span<byte> arg) { }
}
```

Reply:
Please add an item to the breaking change doc.

Reply:
As you mentioned offline, none of the cases above are ambiguous in C#13, and we chose a different overload for `F1()` and `F3()` in C#12 vs. C#13. Updated the example.